index.js

```diff
@@ -207,12 +266,85 @@ class instance extends instance_skel {
 			}
 		}
 
+		let jsonResultDataHandler = (e, result) => {
+			if (e !== null) {
+				this.log('error', `HTTP ${action.action.toUpperCase()} Request failed (${e.message})`)
+				this.status(this.STATUS_ERROR, result.error.code)
+			} else {
+				this.status(this.STATUS_OK)
+
+				debug('processing JSON result.data')
+
+				// retrieve the variable name in which to store json result.data
+				this.system.emit('variable_parse', action.options.jsonResultDataVariable, (value) => {
+					jsonResultDataVariable = value
+				})
+
+				// store json result data into retrieved dedicated custom variable
+				if (jsonResultDataVariable !== '') {
+					debug('jsonResultDataVariable', jsonResultDataVariable)
+					let jsonResultData = JSON.stringify(result.data)
+					this.system.emit('custom_variable_set_value', jsonResultDataVariable, jsonResultData)
+				}
+			}
+		}		
+
 		let options = {
 			connection: {
-				rejectUnauthorized: self.config.rejectUnauthorized,
+				//TODO: verify this trivial fix (replaced 'self' with 'this')
+				rejectUnauthorized: this.config.rejectUnauthorized,
 			},
 		}
 
+// ================================================================================================
+		//TODO: this seems to be better implemented by core-team within internal module,
```

Comment:
should be done with https://github.com/bitfocus/companion-module-generic-http/pull/28/commits/b7c0073f3ab15e4a5c2899fc35b99c21369543ff

Reply:
Possibly to simplify e.g. testing, leave it for now as is? Moving it out should be trivial anyways.

Reply:
After looking a little bit at the companion module, I would prefer to finalize the task as-is. Core-Team would then need to decide where to move the setvar action code, and how to rename the action. Please let me know what you think.

Reply:
(trying to explain a bit: `custom_variable_set` actions seem to need an own module, like `assignments` or `custom_variable`. I would hold all data-assignment stuff. Then, bitfocus-companion code looks to need some refactoring anyways (i see many code-repetitions, and the code re custom_variables is a bit unclear). I would like to avoid to work on this code, even if its more or less just copying the code over)